2020-06-20 20:14:07 [main] WARN  by.mishota.graduation.dao.Dao - Connection isn't closed
java.sql.SQLException: Error releasing connection
	at by.mishota.graduation.dao.pool.ProxyConnection.close(ProxyConnection.java:65) ~[classes/:?]
	at by.mishota.graduation.dao.Dao.close(Dao.java:35) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.findStudents(StudentDaoImpl.java:104) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.findById(StudentDaoImpl.java:67) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.main(StudentDaoImpl.java:77) [classes/:?]
Caused by: by.mishota.graduation.exception.ConnectionPoolException: Can't return the connection to the pool that is not from the pool
	at by.mishota.graduation.dao.pool.ConnectionPool.releaseConnection(ConnectionPool.java:115) ~[classes/:?]
	at by.mishota.graduation.dao.pool.ProxyConnection.close(ProxyConnection.java:63) ~[classes/:?]
	... 4 more
2020-06-20 20:21:23 [main] WARN  by.mishota.graduation.dao.Dao - Connection isn't closed
java.sql.SQLException: Error releasing connection
	at by.mishota.graduation.dao.pool.ProxyConnection.close(ProxyConnection.java:65) ~[classes/:?]
	at by.mishota.graduation.dao.Dao.close(Dao.java:35) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.findStudents(StudentDaoImpl.java:104) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.findById(StudentDaoImpl.java:67) [classes/:?]
	at by.mishota.graduation.dao.impl.StudentDaoImpl.main(StudentDaoImpl.java:77) [classes/:?]
Caused by: by.mishota.graduation.exception.ConnectionPoolException: Can't return the connection to the pool that is not from the pool
	at by.mishota.graduation.dao.pool.ConnectionPool.releaseConnection(ConnectionPool.java:115) ~[classes/:?]
	at by.mishota.graduation.dao.pool.ProxyConnection.close(ProxyConnection.java:63) ~[classes/:?]
	... 4 more
2020-08-17 11:38:35 [main] INFO  by.mishota.graduation.dao.EntityTransaction - Rollback is error
java.sql.SQLNonTransientConnectionException: Can't call rollback when autocommit=true
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:110) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:1833) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at by.mishota.graduation.dao.pool.ProxyConnection.rollback(ProxyConnection.java:56) ~[classes/:?]
	at by.mishota.graduation.dao.EntityTransaction.rollback(EntityTransaction.java:28) ~[classes/:?]
	at by.mishota.graduation.dao.impl.EntrantDaoImpl.addResult(EntrantDaoImpl.java:248) ~[classes/:?]
	at by.mishota.graduation.dao.impl.EntrantDaoImplTest.testAddResultOnNull(EntrantDaoImplTest.java:114) ~[test-classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:134) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:597) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173) ~[testng-7.1.0.jar:?]
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:816) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146) ~[testng-7.1.0.jar:?]
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128) ~[testng-7.1.0.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1510) [?:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:766) [testng-7.1.0.jar:?]
	at org.testng.TestRunner.run(TestRunner.java:587) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunner.run(SuiteRunner.java:286) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53) [testng-7.1.0.jar:?]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96) [testng-7.1.0.jar:?]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1187) [testng-7.1.0.jar:?]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1109) [testng-7.1.0.jar:?]
	at org.testng.TestNG.runSuites(TestNG.java:1039) [testng-7.1.0.jar:?]
	at org.testng.TestNG.run(TestNG.java:1007) [testng-7.1.0.jar:?]
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66) [testng-rt.jar:?]
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109) [testng-rt.jar:?]
